File: xiami.py
#encoding:utf-8
import urllib


def caesar(location):
    num = int(location[0])
    avg_len, remainder = int(len(location[1:]) / num), int(len(location[1:]) % num)
    result = [location[i * (avg_len + 1) + 1: (i + 1) * (avg_len + 1) + 1] for i in range(remainder)]
    result.extend([location[(avg_len + 1) * remainder:][i * avg_len + 1: (i + 1) * avg_len + 1] for i in range(num-remainder)])
    url = urllib.unquote(''.join([''.join([result[j][i] for j in range(num)]) for i in range(avg_len)]) + \
                        ''.join([result[r][-1] for r in range(remainder)])).replace('^','0')
    return url

File: xiami.js
module.export = (location) => {
    let num = location[0];
    let params = location.substr(1, location.length);

    let sliceLength = Math.floor(params.length / num);

    let remainLength = params.length % num;
    let group = [];
    let result = '';

    for (let i=0; i<num; ++i) {
        group[i] = i<remainLength ? params.substr((sliceLength+1)*i, sliceLength+1) : params.substr(sliceLength * (i - remainLength) + (sliceLength + 1) * remainLength, sliceLength);
    }

    for (let i=0; i<group[0].length; ++i) {
        for (let j=0; j<group.length; ++j) {
            result += group[j][i] ? group[j][i] : '';
        }
    }

    return decodeURIComponent(result).replace(/\^/g, '0');
};
